src/libraries/System.Net.Http/tests/UnitTests/Headers/ObjectCollectionTest.cs

```diff
@@ -36,9 +36,23 @@ public void Ctor_ExecuteBothOverloads_MatchExpectation()
             c.Add("value1");
 
             Assert.Throws<InvalidOperationException>(() => { c.Add(null); });
+        }
+
+        [Fact]
+        public void TestContainsAgainstNonSingleOrEmpty()
+        {
+            // Use default validator
+            ObjectCollection<string> c = new ObjectCollection<string>();
+
+            Assert.Equal(0, c.Count);
+            Assert.False(c.Contains("value" + 1));
+
+            c.Add("value" + 1);
 
             Assert.Equal(1, c.Count);
-            Assert.True(c.Contains("value1"));
+            // Force the reference to be different to ensure we are checking for semantic equality
+            // and not reference equality.
+            Assert.True(c.Contains("value" + 1));
```

Comment:
Can you also test what happens when you add another item and then remove?  At that point the _size will be 1 but _items will be an array.  The original code was trying to avoid a type check in the case of the collection containing only one element, but I'm wondering if it'd be better to just simplify the logic to be more like:
```
public bool Contains(T item) =>
    _items is T[] items ? Array.IndexOf(items, item, 0, _size) != -1 :
    _items is object o ? o.Equals(item) :
    item is null;
```
That may or may not be slightly more expensive, but I think it's also easier to reason about, e.g. in the current PR, we'll end up evaluating both conditions if the _size is 1 but _items is an array because items were added and then removed.

Reply:
We should also add tests for when the item isn't the first one but rather exists later in the collection.  And for when the item can't be found in a collection that contains multiple entries.

Reply:
The prior implementation `_items` and `item` being null was to return true. Do we really want to return true there?

Reply:
No,  if the collection is empty,  it should return false